diesel/src/pg/expression/functions.rs

```diff
@@ -1423,3 +1423,38 @@ define_sql_function! {
     /// ```
     fn array_upper<Arr: ArrayOrNullableArray + SingleValue>(array: Arr, dimension: Integer) -> Nullable<Integer>;
 }
+
+define_sql_function! {
+    /// Builds a JSON object out of a text array. The array must have an even number of members,
+    /// in which case they are taken as alternating key/value pairs
+    ///
+    /// # Example
+    ///
+    /// ```rust
+    /// # include!("../../doctest_setup.rs");
+    /// #
+    /// # fn main() {
+    /// #     run_test().unwrap();
+    /// # }
+    /// #
+    /// # fn run_test() -> QueryResult<()> {
+    /// #     use diesel::dsl::json_object;
+    /// #     use diesel::sql_types::{Json};
+    /// #     use serde_json::Value;
+    /// #     let connection = &mut establish_connection();
+    /// let json = diesel::select(json_object(vec!["hello","world"])).get_result::<Value>(connection)?;
+    /// let expected:Value = serde_json::from_str(r#"{"hello":"world"}"#).unwrap();
+    /// assert_eq!(expected,json);
+    ///
+    /// let json = diesel::select(json_object(vec!["hello","world","John","Doe"])).get_result::<Value>(connection)?;
+    /// let expected:Value = serde_json::from_str(r#"{"hello":"world","John":"Doe"}"#).unwrap();
+    /// assert_eq!(expected,json);
+    ///
+    /// let json = diesel::select(json_object(vec!["hello","world","John"])).get_result::<Value>(connection);
+    /// assert!(json.is_err());
+    ///
+    /// #     Ok(())
+    /// # }
+    /// ```
+    fn json_object(text_array:Array<Text>)->Json;
```

Comment:
Array are always generated with nullable elements in diesel which can make hard to use this function. I think this functions should be able to accept `Nullable<Array<Nullable<Text>>`, maybe using a helper trait to make simple to call it directly.

Reply:
I would just accept both variants here. That might require an additional helper trait that is implemented for both `Array<Text>` and `Array<Nullable<Text>>`.  Additionally what's about `Nullable<Array<_>>` here?